ecc/bls12-377/ecdsa/marshal.go

```diff
@@ -97,9 +109,22 @@ func (sig *Signature) Bytes() []byte {
 // It returns the number of bytes read from buf.
 func (sig *Signature) SetBytes(buf []byte) (int, error) {
 	n := 0
-	if len(buf) < sizeSignature {
-		return n, io.ErrShortBuffer
+	if len(buf) != sizeSignature {
+		return n, ErrWrongSizeBuffer
 	}
+
+	// S, R < R_mod (to avoid malleability)
+	frMod := fr.Modulus()
+	var bufBigInt big.Int
+	bufBigInt.SetBytes(buf[:sizeFr])
+	if bufBigInt.Cmp(frMod) != -1 {
+		return 0, ErrRBiggerThanRMod
```

Comment:
Similarly, imo is better to use `return 0, fmt.Errorf("r >= Fr")` and remove constant error value.